File: Assets/Scripts/Window/PanelAlert.cs
using UnityEngine;
using System.Collections;
using UnityEngine.UI;

public enum AlertRet {
	OK=0,
	CANCEL,
}
public delegate void AlertCallback(AlertRet ret);

public class PanelAlert : BaseWindow {
	private RectTransform btn_ok;
	private RectTransform btn_cancel;

	private Text text_title;
	private Text text_msg;

	public AlertCallback _callback;

	void Awake () {
		btn_ok = transform.FindChild("Button Ok") as RectTransform;
		btn_cancel = transform.FindChild("Button Cancel") as RectTransform;

		text_title = transform.FindChild("Text Title").GetComponent<Text>();
		text_msg = transform.FindChild("Text Msg").GetComponent<Text>();
	}

	public override void Init(object data) {
		transform.localPosition = new Vector3(0, 0, -1000);
	}

	public void SetTitle(string title) {
		text_title.text = title;
	}
	public void SetMsg(string msg) {
		text_msg.text = msg;
	}
	public void SetCallback(AlertCallback callback) {
		_callback = callback;
	}

	void Start() {
		if (_callback == null) {
			btn_ok.anchoredPosition = new Vector2(0, btn_ok.anchoredPosition.y);
			btn_cancel.gameObject.SetActive(false);
		}
	}

	public void OnBtnClose() {
		if (_callback != null) _callback(AlertRet.CANCEL);
		GUIManager.GetInstance().HideAlert();
		SoundManager.instance.PlaySE(SoundManager.UI_BTN);
	}

	public void OnBtnOk() {
		if (_callback != null) _callback(AlertRet.OK);
		GUIManager.GetInstance().HideAlert();
		SoundManager.instance.PlaySE(SoundManager.UI_BTN);
	}

	public void OnBtnCancel() {
		if (_callback != null) _callback(AlertRet.CANCEL);
		GUIManager.GetInstance().HideAlert();
		SoundManager.instance.PlaySE(SoundManager.UI_BTN);
	}
}

File: Assets/Lua/Window/PanelAlert.lua
local gameObject
local transform

PanelAlert = {}
local this = PanelAlert
this._name = 'PanelAlert'
local TAG = 'PanelAlert'

-- --------------------------------------------------------------------
--	c# callback
-----------------------------------------------------------------------
function PanelAlert.Awake(obj)
	gameObject = obj
	transform = obj.transform
end

function PanelAlert.OnDestroy()
	gameObject = nil
	this.bInit = false
end

-- --------------------------------------------------------------------
--	mvc notication
-----------------------------------------------------------------------
function PanelAlert:listNotificationInterests()
	return {}
end
function PanelAlert:handleNotification(notification)
end
function PanelAlert:init(...)
	this.title, this.msg, this.callback = ...

	print (TAG, "init", this.title, this.msg, this.callback)

	this.InitPanel()
end


--初始化面板--
this.bInit = false
function PanelAlert.InitPanel()
	print(TAG, 'InitPanel', this.bInit)
	this.btn_ok = transform:FindChild("Button OK").gameObject
	this.btn_cancel = transform:FindChild("Button Cancel").gameObject

	local text_title = transform:FindChild("Text Title"):GetComponent('Text')
	local text_msg = transform:FindChild("Text Msg"):GetComponent("Text")
	text_title.text = this.title
	text_msg.text = this.msg

	if this.bInit == false then
		window = transform:GetComponent('LuaBehaviour')
		window:AddClick(this.btn_ok, this.OnBtnOK)
		window:AddClick(this.btn_cancel, this.OnBtnCancel)
	end
	this.bInit = true
end


-- --------------------------------------------------------------------
--	click event
-----------------------------------------------------------------------
function PanelAlert.OnBtnOK(go)
	print('OnBtnOK')
	guiMgr:HideWindow(gameObject)
	if this.callback then
		this.callback(1)
	end
end

function PanelAlert.OnBtnCancel(go)
	print('OnBtnCancel')
	guiMgr:HideWindow(gameObject)	
	if this.callback then
		this.callback(0)
	end
end

return PanelAlert